### PR TITLE
Add support for `Int128` in codegen and macros

### PR DESCRIPTION
**Pull request by BlobCodes**
This PR adds:
- Ability to use U/Int128 in macros
- MathInterpreter U/Int128 support (for ENUMs and compile-time interpreted generics)
- U/Int128 LLVM inlining (probably improves performance?)

Things that still need to be addressed: 
- The macro `to_i` method always converts a string to a int64, and then casts it to a int32. integers greater than Int64::MAX return an error "(number) is not an integer", while integers greater than Int32::MAX but smaller than or equal Int64::MAX return an error "Invalid Int32: (number).
- Crystal::Call#convert_numeric_argument does not seem to support int128, but I don't know what it is used for.

Probably requires #11551 to pass compiler specs on windows.

### CODE REVIEW

File: spec/compiler/macro/macro_methods_spec.cr [hunk 1]
```diff
@@ -3029,4 +3042,11 @@ module Crystal
       assert_macro %({{ (x.conds << "a"; x.conds.size) }}), "2", {x: node}
     end
   end
+
+  describe "int128 support" do
+    it "" do
+      node = When.new([2.int32, 3.int32] of ASTNode, 4.int32)
+      assert_macro %({{ (x.conds << "a"; x.conds.size) }}), "2", {x: node}
+    end
+  end
```
Comment (straight-shoota) [inline spec/compiler/macro/macro_methods_spec.cr:3051]: What does this example do with 128-bit integers?

File: spec/compiler/macro/macro_methods_spec.cr [hunk 2]
```diff
@@ -276,12 +277,24 @@ module Crystal
...
       it "#to_number" do
         assert_macro "{{ 4_u8.to_number }}", "4"
         assert_macro "{{ 2147483648.to_number }}", "2147483648"
         assert_macro "{{ 1_f32.to_number }}", "1.0"
+        assert_macro "{{ 4_u128.to_number }}", "4"
+        assert_macro "{{ -20i128.to_number }}", "-20"
+      end
+
+      # Requires #11571
+      pending "executes math operations using U/Int128" do
```
Comment (straight-shoota) [inline spec/compiler/macro/macro_methods_spec.cr:293]: Just leaving this comment as a reminder to remove the `pending` after #11571

File: spec/compiler/macro/macro_methods_spec.cr (hunk 1 above)
Comment (BlobCodes) [reply spec/compiler/macro/macro_methods_spec.cr:3051]: Oops, looks like I just copied this spec:
https://github.com/crystal-lang/crystal/blob/baf5c3b3045c8401c4364a6e7f351d314b1e1fb5/spec/compiler/macro/macro_methods_spec.cr#L3027-L3030

File: spec/compiler/macro/macro_methods_spec.cr (hunk 2 above)
Comment (BlobCodes) [reply spec/compiler/macro/macro_methods_spec.cr]: done in c813d0550f68ea0f7ccdab0ff5d7fb1f7406cadc

**Comment by oprypin** [comment]
The CI failures look real (related to this pull request).

**Comment by watzon** [comment]
Seems like the only failing specs currently are in the interpreter due to a missing `Syscall::GRND_NONBLOCK`. Does this still seem related?

**Review by straight-shoota — APPROVED** [review]

**Review by oprypin — APPROVED** [review]

**Pull request merged**